### PR TITLE
Better deployment

### PR DESCRIPTION
**Pull request by gbenhaim**
See the different commits for the improvements added for the deployment.

### CODE REVIEW

**Comment by yftacherzog** [comment]
When deploying for the first time, we encountered an issue with open-files limit.
Maybe we can mention that and a way to work around it (`sudo sysctl fs.inotify.max_user_watches=524288`, `sudo sysctl fs.inotify.max_user_instances=512`)

**Comment by avi-biton** [comment]
When I follow the steps, I get some errors:
`error: timed out waiting for the condition on tektonconfigs/config` 
when I ran the `deploy-deps.sh` script

`Error from server (BadRequest): container "copy-chrome-static-content" in pod "proxy-5f6c6f89c6-hwm67" is waiting to start: PodInitializing` 
when I ran the `deploy-konflux.sh` script.

and we can see that the proxy pod is failing:
`proxy-5f6c6f89c6-hwm67               1/3     CrashLoopBackOff   8 (33s ago)   11m`

And the log:
```
Defaulted container "nginx-120" out of: nginx-120, workspace-manager, oauth2-proxy, copy-chrome-static-content (init), add-sva-token-to-nginx-config (init)
2024/04/17 12:44:28 [emerg] 1#1: host not found in upstream "keycloak-service.keycloak.svc.cluster.local" in /etc/nginx/nginx.conf:67
nginx: [emerg] host not found in upstream "keycloak-service.keycloak.svc.cluster.local" in /etc/nginx/nginx.conf:67
```

**Comment by gbenhaim** [comment]
@avi-biton did you checked out the code in this pr?

**Comment by avi-biton** [comment]
> @avi-biton did you checked out the code in this pr?
Yes I did

**Comment by yftacherzog** [comment]
Let's add openssl as prerequisite in readme

File: README.md [hunk 1]
```diff
@@ -7,16 +7,34 @@ The recommended way to try out Konflux is using [Kind](https://kind.sigs.k8s.io/
 Create a Kind cluster using the provided config in this repository.
 The config tells Kind to forward port `9443` from the host to the Kind cluster. The port forwarding is needed for accessing Konflux.
 
-From the root of this repository, run the following commands:
+### Machine requirements
+
+Th deployment requires 8GB of free RAM.
```
Comment (yftacherzog) [inline README.md]:
```suggestion
The deployment requires 8GB of free RAM.
```

Comment (gbenhaim) [reply README.md]: done

**Pull request merged**